### ISSUE TITLE
HEAD requests for a partNumber no longer return a ranged reply

### ISSUE THREAD

**Issue by jt121dot5** [minio/minio#11989]
A regression between RELEASE.2020-11-13T20-10-18Z and RELEASE.2021-03-26T00-00-41Z has changed the Content information returned by part-specific HEAD requests.

This might be the same issue as #11824

## Expected Behavior

An HTTP request for part number 1: 

>HEAD /repo1/img?partNumber=1 HTTP/1.1
> if-match: "ccb718aa1f39f6416f789cf733a387c7-7"
> x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
> x-amz-date: 20210407T003436Z
> ...

should return a ranged response covering the range of bytes for the requested part:

> HTTP/1.1 206 Partial Content
> Accept-Ranges: bytes
> Content-Length: 59768832
> Content-Range: bytes 0-59768831/417988608
> Content-Security-Policy: block-all-mixed-content
> Content-Type: application/octet-stream
> ETag: "ccb718aa1f39f6416f789cf733a387c7-7"
> Last-Modified: Tue, 06 Apr 2021 20:48:30 GMT
> Server: MinIO/RELEASE.2020-11-13T20-10-18Z
> Vary: Origin
> X-Amz-Request-Id: 16736C36CF553906
> X-Xss-Protection: 1; mode=block
> x-amz-mp-parts-count: 7
> Date: Wed, 07 Apr 2021 00:34:33 GMT

## Current Behavior

Reply now covers the whole file, rather than just the bytes in the requested part:

> HTTP/1.1 206 Partial Content <
> Accept-Ranges: bytes <
> Content-Length: 417988608 <
> Content-Range: bytes 0-417988607/417988608 <
> Content-Security-Policy: block-all-mixed-content <
> Content-Type: application/octet-stream <
> ETag: "ccb718aa1f39f6416f789cf733a387c7-7" <
> Last-Modified: Mon, 05 Apr 2021 19:13:29 GMT <
> Server: MinIO <
> Vary: Origin <
> X-Amz-Request-Id: 1673686323A60D1F <
> X-Xss-Protection: 1; mode=block <
> x-amz-mp-parts-count: 7 <
> Date: Tue, 06 Apr 2021 23:24:26 GMT

## Steps to Reproduce (for bugs)

1. upload a multi-part file
2. issue a HEAD request for partNumber=1 - reply should cover the range of part number 1, not the whole file

## Context

While this particular issue is one that I can work around, I'm posting it here because current behavior contradicts AWS S3 documentation and so will affect any applications that depend on the proper Content-range information being returned. 

## Regression
Is this issue a regression? (Yes / No) Yes

**Comment by harshavardhana** [comment]
This is present in erasure-coded setup

```
curl "http://localhost:9000/testbucket/ubuntu-18.04-desktop-amd64.iso?partNumber=2" -o /dev/null -v
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 127.0.0.1:9000...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 9000 (#0)
> GET /testbucket/ubuntu-18.04-desktop-amd64.iso?partNumber=2 HTTP/1.1
> Host: localhost:9000
> User-Agent: curl/7.68.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 206 Partial Content
< Accept-Ranges: bytes
< Content-Length: 1921843200
< Content-Range: bytes 0-1921843199/1921843200
< Content-Security-Policy: block-all-mixed-content
< Content-Type: application/x-iso9660-image
< ETag: "60f529b0a77763d3f262f2d723d1968f-115"
< Last-Modified: Wed, 07 Apr 2021 05:51:44 GMT
< Server: MinIO
< Vary: Origin
< X-Amz-Request-Id: 16737D94812F97C3
< X-Amz-Server-Side-Encryption: AES256
< X-Xss-Protection: 1; mode=block
< x-amz-mp-parts-count: 115
< Date: Wed, 07 Apr 2021 05:52:47 GMT
< 
{ [3583 bytes data]
* transfer closed with 1905065984 bytes remaining to read
  0 1832M    0 16.0M    0     0   104M      0  0:00:17 --:--:--  0:00:17  104M
* Closing connection 0
curl: (18) transfer closed with 1905065984 bytes remaining to read
```

**Comment by harshavardhana** [comment]
and in the FS mode as well works fine

```
curl "http://localhost:9000/testbucket/ubuntu-18.04-desktop-amd64.iso?partNumber=2" -o /dev/null -v
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 127.0.0.1:9000...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 9000 (#0)
> GET /testbucket/ubuntu-18.04-desktop-amd64.iso?partNumber=2 HTTP/1.1
> Host: localhost:9000
> User-Agent: curl/7.68.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 206 Partial Content
< Accept-Ranges: bytes
< Content-Length: 1921843200
< Content-Range: bytes 0-1921843199/1921843200
< Content-Security-Policy: block-all-mixed-content
< Content-Type: application/x-iso9660-image
< ETag: "483ecf67aa9ead01210c819b3718c2f9-115"
< Last-Modified: Wed, 07 Apr 2021 05:54:22 GMT
< Server: MinIO
< Vary: Origin
< X-Amz-Request-Id: 16737DAD09FA5354
< X-Amz-Server-Side-Encryption: AES256
< X-Xss-Protection: 1; mode=block
< x-amz-mp-parts-count: 115
< Date: Wed, 07 Apr 2021 05:54:33 GMT
< 
{ [3583 bytes data]
* transfer closed with 1905065984 bytes remaining to read
  0 1832M    0 16.0M    0     0   168M      0  0:00:10 --:--:--  0:00:10  168M
* Closing connection 0
curl: (18) transfer closed with 1905065984 bytes remaining to read
```

Correct content-rage is returned for a multipart object with 115 parts

**Comment by harshavardhana** [comment]
You are right the returned content-range is wrong
